xdsl/dialects/x86/ops.py

```diff
@@ -276,6 +285,112 @@ class MovOp(RROperation[GeneralRegisterType, GeneralRegisterType]):
     name = "x86.mov"
 
 
+class ROperation(Generic[R1InvT], SingleOperandInstruction):
+    """
+    A base class for x86 operations that have one register.
+    """
+
+    source = opt_operand_def(R1InvT)
+    destination = opt_result_def(R1InvT)
+
+    def __init__(
+        self,
+        source: Operation | SSAValue | None = None,
+        *,
+        comment: str | StringAttr | None = None,
+        destination: R1InvT | None = None,
+    ):
+        if isinstance(comment, str):
+            comment = StringAttr(comment)
+
+        super().__init__(
+            operands=[source],
+            attributes={
+                "comment": comment,
+            },
+            result_types=[destination],
+        )
+
+    def verify_(self) -> None:
+        if self.source is None and self.destination is None:
+            raise VerifyException("Either source or destination must be specified")
+        return super().verify_()
+
+    def assembly_line_args(self) -> tuple[AssemblyInstructionArg | None, ...]:
+        return (self.destination,) if self.destination else (self.source,)
+
+
+@irdl_op_definition
+class IdivOp(ROperation[GeneralRegisterType]):
```

Comment:
IMO, in a perfect world, I would love it if you could encode the fact that rax and rdx are the needed registers in a declarative way. I imagine something like:

```python
class IdivOp:

  divisend = operand_def(x86RegisterType("rdx"))
  divisor = operand_def(x86RegisterType("rax"))
  quotient = ...
  remainder = ...
```

I am honestly not sure if this will be possible in the current xDSL infrastructure and with the way you have set up the x86Registertype. But it would still be cool to do somehow :sweat_smile: 

IMO, it might make sense to remove Idiv from this PR and submit one for this specifically, then we could have this discussion there and not stall this one :)